### PR TITLE
Update `react-replace-text-components` migration

### LINKED ISSUE #7786: [Typography foundations] Update migration to insert comments when not JSXIdentifiers/strings

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Resolves #7786.
Updates typography migration to insert comments for complex text migrations.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by laurkim** [comment]
Added the `Skip changelog` label temporarily so that other CI checks run.

**Review by laurkim — APPROVED** [review]

File: polaris-migrator/src/migrate.ts [hunk 1]
```diff
@@ -48,7 +48,6 @@ export async function migrate(
       extensions: 'tsx,ts,jsx,js',
       parser: 'tsx',
       verbose: 2,
-      runInBand: true,
```
Comment (sam-b-rose) [inline polaris-migrator/src/migrate.ts:51]: nice!! ⚡

**Review by sam-b-rose — APPROVED** [review]

**Comment by sam-b-rose** [comment]
Is this ready to be merged into `main`? @aaronccasanova

**Comment by aaronccasanova** [comment]
@samrose Thanks for the follow up. Lo mentioned before the holiday break that there are certain cases where legacy typography components are nested and replacements are causing styling/cascade issues. Before merging, I think we need to understand the issue further and decide if we can fix it programmatically or if we need to document a manual workflow.

**Comment by samrose** [comment]
@samrose3 they were looking for you ^^^

**Comment by sam-b-rose** [comment]
After trying out the snapshot of this migration on web, I noticed an edge-case we are missing with nested `TextStyle` components and injecting `InlineCode`:

```tsx
<TextStyle>
  This is nested <TextStyle variation="code">InlineCode</TextStyle>
</TextStyle>
```

The above will not properly inject the `InlineCode` component. I've tried adjusting the migration script to account for this scenario but without luck. This is something we'll likely want to address or at least add a manual check comment.

**Comment by aaronccasanova** [comment]
Good catch @samrose3! The issue was the `removeJSXAttributes` util was removing `variation` attributes deeply. Here's my WIP fix: https://github.com/Shopify/polaris/pull/7783/commits/5783997f05e2e3e85fddccfcc1767a6fd8fa4f60

**Pull request merged**